src/main/java/jeff/CommandHandler.java

```diff
@@ -0,0 +1,123 @@
+package jeff;
+
+import jeff.exceptions.*;
+
+public class CommandHandler {
+    private static final int TODO_DESCRIPTION_INDEX = 5;
+    private static final int DEADLINE_DESCRIPTION_INDEX = 9;
+    private static final int DEADLINE_BY_INDEX_LENGTH = 4;
+    private static final int EVENT_DESCRIPTION_INDEX = 6;
+    private static final int EVENT_FROM_INDEX_LENGTH = 6;
+    private static final int EVENT_TO_INDEX_LENGTH = 4;
+    private static final int MARK_INDEX = 5;
+    private static final int UNMARK_INDEX = 7;
+
+    public static void handleList() {
+        Printer.printTasks();
+    }
+
+    public static void handleTodo(String userInput) throws InvalidTodoSyntaxException {
+        if (userInput.equals("todo")) {
+            throw new InvalidTodoSyntaxException();
+        }
+        String description = userInput.substring(TODO_DESCRIPTION_INDEX);
+        Todo todo = new Todo(description);
+        TaskList.add(todo);
+        FileManager.appendTask(todo);
+        Printer.printAddTask();
+    }
+
+    public static void handleDeadline(String userInput) throws InvalidDeadlineSyntaxException {
+        int byIndex = userInput.indexOf(" /by ") + 1;
+        if (byIndex == 0) {
+            throw new InvalidDeadlineSyntaxException();
+        }
+        String description = userInput.substring(DEADLINE_DESCRIPTION_INDEX, byIndex);
+        String by = userInput.substring(byIndex + DEADLINE_BY_INDEX_LENGTH);
+        if (description.isEmpty()) {
+            throw new InvalidDeadlineSyntaxException();
+        }
+        Deadline deadline = new Deadline(description, by);
+        TaskList.add(deadline);
+        FileManager.appendTask(deadline);
+        Printer.printAddTask();
+    }
+
+    public static void handleEvent(String userInput) throws InvalidEventSyntaxException {
```

Comment:
spacing in the functions can be improved.

https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#structure-code-logically